x-pack/plugins/transform/public/app/sections/transform_management/components/transform_list/action_delete.tsx

```diff
@@ -71,17 +85,96 @@ export const DeleteAction: FC<DeleteActionProps> = ({ items, forceDisable }) =>
     defaultMessage: 'Delete {transformId}',
     values: { transformId: items[0] && items[0].config.id },
   });
-  const bulkDeleteModalMessage = i18n.translate(
-    'xpack.transform.transformList.bulkDeleteModalBody',
-    {
-      defaultMessage:
-        "Are you sure you want to delete {count, plural, one {this} other {these}} {count} {count, plural, one {transform} other {transforms}}? The transform's destination index and optional Kibana index pattern will not be deleted.",
-      values: { count: items.length },
-    }
+  const bulkDeleteModalContent = (
+    <>
+      <p>
+        <FormattedMessage
+          id="xpack.transform.transformList.bulkDeleteModalBody"
+          defaultMessage="Are you sure you want to delete {count, plural, one {this} other {these}} {count} {count, plural, one {transform} other {transforms}}?"
+          values={{ count: items.length }}
+        />
+      </p>
+      <EuiFlexGroup direction="column" gutterSize="none">
+        <EuiFlexItem>
+          {
+            <EuiSwitch
+              data-test-subj="transformBulkDeleteIndexSwitch"
+              style={{ paddingBottom: 10 }}
+              label={i18n.translate(
+                'xpack.transform.actionDeleteTransform.bulkDeleteDestinationIndexTitle',
+                {
+                  defaultMessage: 'Delete destination indexes',
```

Comment:
nit: for plural, it's `indices`, at least for I observed in the documentation

Reply:
Good spot from @darnautov! Could you change it to `indices` here @qn895 , and also for the data frame analytics page too if that went in at `indexes` too.